annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -589,6 +592,35 @@ private String solrUniquenessQueryLinking(Annotation anno) {
 		solrQuery+=")";
 		return solrQuery;
 	}
+	
+	private String solrUniquenessQueryLinkForContributing(Annotation anno) {
+      String solrQuery = "(" + WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.LINKFORCONTRIBUTING.getOaType() + "\"";
+      
+      List<String> targetValues = anno.getTarget().getValues();
+      if(targetValues!=null) {
+        for (String targetValue : targetValues) {
+            solrQuery+=" AND " + SolrAnnotationConstants.TARGET_URI + ":\"" + targetValue + "\"";
+        }
+      }
+      
+      if(anno.getTarget().getValue()!=null) {
+        solrQuery+=" AND " + SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"";
+      }
+
+      List<String> bodyUris = extractUriValues(anno.getBody());
+      if(bodyUris!=null) {
+        for (String bodyUri : bodyUris) {
+            solrQuery+=" AND " + SolrAnnotationConstants.BODY_URI + ":\"" + bodyUri + "\""; 
+        }
+      }
+      
+      if(anno.getBody().getValue()!=null) {
```

Comment:
body URIs are extracted from the body, this criterion is redundant and must be removed

Reply:
Done.